lua/nvim-lsp-installer/servers/omnisharp/init.lua

```diff
@@ -7,55 +7,83 @@ local std = require "nvim-lsp-installer.core.managers.std"
-                        when(
-                            platform.is_mac,
-                            coalesce(
-                                when(platform.arch == "x64", "omnisharp-osx-x64-net6.0.zip"),
-                                when(platform.arch == "arm64", "omnisharp-osx-arm64-net6.0.zip")
-                            )
-                        ),
-                        when(
-                            platform.is_linux,
-                            coalesce(
-                                when(platform.arch == "x64", "omnisharp-linux-x64-net6.0.zip"),
-                                when(platform.arch == "arm64", "omnisharp-linux-arm64-net6.0.zip")
-                            )
-                        ),
-                        when(
-                            platform.is_win,
-                            coalesce(
-                                when(platform.arch == "x64", "omnisharp-win-x64-net6.0.zip"),
-                                when(platform.arch == "arm64", "omnisharp-win-arm64-net6.0.zip")
-                            )
-                        )
-                    ),
-                }).with_receipt()
-            end)
-        end,
-        default_options = {
-            cmd = {
-                "dotnet",
-                path.concat { root_dir, "omnisharp", "OmniSharp.dll" },
-                "--languageserver",
-                "--hostPID",
-                tostring(vim.fn.getpid()),
-            },
-        },
-    }
+      -- We write to the omnisharp directory for backwards compatibility reasons
+      ctx.fs:mkdir "omnisharp"
+      ctx:chdir("omnisharp", function()
+        github.unzip_release_file({
+          repo = "OmniSharp/omnisharp-roslyn",
+          asset_file = coalesce(
+            when(
+              platform.is_mac,
+              coalesce(
+                when(platform.arch == "x64", "omnisharp-osx-x64-net6.0.zip"),
+                when(platform.arch == "arm64", "omnisharp-osx-arm64-net6.0.zip")
+              )
+            ),
+            when(
+              platform.is_linux,
+              coalesce(
+                when(platform.arch == "x64", "omnisharp-linux-x64-net6.0.zip"),
+                when(platform.arch == "arm64", "omnisharp-linux-arm64-net6.0.zip")
+              )
+            ),
+            when(
+              platform.is_win,
+              coalesce(
+                when(platform.arch == "x64", "omnisharp-win-x64-net6.0.zip"),
+                when(platform.arch == "arm64", "omnisharp-win-arm64-net6.0.zip")
+              )
+            )
+          ),
+        }).with_receipt()
+      end)
+
+      ctx.fs:mkdir "omnisharp-mono"
+      ctx:chdir("omnisharp-mono", function()
+        github.unzip_release_file({
+          repo = "OmniSharp/omnisharp-roslyn",
+          asset_file = "omnisharp-mono.zip",
+        }).with_receipt()
+      end)
+    end,
+    default_options = {
+      cmd = GenerateCmd(nil, root_dir),
```

Comment:
Since we're always overwriting the `config.cmd` inside `on_new_config` (this is called by lspconfig whenever it realizes it needs to start the new server), this initial `cmd` value is actually not needed:

```suggestion
```